shopgate.php

```diff
@@ -791,40 +844,49 @@ public function getContent()
 				$languageID = Language::getIdByIso($configs['language']);
 			Configuration::updateValue('SHOPGATE_LANGUAGE_ID', $languageID);
 
-			try {
+			try
+			{
 				$shopgateConfig->loadArray($configs);
 				$shopgateConfig->initFolders();
 				$shopgateConfig->save(array_keys($configs));
 				$output .= '<div class="conf confirm"><img src="../img/admin/ok.gif" alt="'.$this->l('Confirmation').'" />'.$this->l('Configurations updated').'</div>';
-				Configuration::updateValue('SHOPGATE_CONFIGURATION_OK', true);
-			} catch (ShopgateLibraryException $e) {
+			} catch(ShopgateLibraryException $e)
+			{
 				$output .= '<div class="conf error"><img src="../img/admin/error.png" alt="'.$this->l('Error').'" />'.$this->l('Error').': '.$e->getAdditionalInformation().'</div>';
 			}
 		}
 
-		$langs = array();
+		$langs = array ();
 		foreach (Language::getLanguages() as $id => $l)
 			$langs[Tools::strtoupper($l['iso_code'])] = $l['name'];
 
-		$servers = array(
-			'live' => $this->l('Live'),
-			'pg' => $this->l('Playground'),
+		$servers = array (
+			'live'   => $this->l('Live'),
+			'pg'     => $this->l('Playground'),
 			'custom' => $this->l('Custom')
 		);
 
-		$enables = array();
+		$enables = array ();
 
 		$settings = Configuration::getMultiple($settingKeys);
 		/**
 		 * read config from db
 		 */
 		$shopgateConfig = new ShopgateConfigPresta(
-			Configuration::get('SHOPGATE_CONFIG') ?
+			is_array(unserialize(Configuration::get('SHOPGATE_CONFIG'))) ?
 				unserialize(Configuration::get('SHOPGATE_CONFIG')) :
-				array()
+				array ()
```

Comment:
Here again, please store the result in a temporary variable instead of unserializing two times